### ISSUE TITLE
Use Git Data API for files

### ISSUE THREAD

**Issue by Rexogamer** [gitpoint/git-point#889]
This would allow us to get files up to 100mb in size.

**Comment by chinesedfan** [comment]
Thanks for your suggestion. But how do users view such large files? It is hard to navigate to certain lines or do anything else.

**Comment by Rexogamer** [comment]
....what? I don’t really see what that has to do with this...

**Comment by chinesedfan** [comment]
I mean files that are larger than 1mb must contain thousands or more lines of codes. So even if we load them, users can't read them.

Can you provide a large file example? Like the file path in a repository. And what do you expect for which screen? The file list or the file detail?

**Comment by Rexogamer** [comment]
This is simply an API change. No UI changes were requested in this issue, if that was what you were asking.

[When I try to load this file,](https://github.com/DJScias/Discord-Datamining/blob/master/current.js), I get the following error:
![794A202B-D2B8-47AD-9B18-81BC13E479F0](https://user-images.githubusercontent.com/42586271/65779675-1fd94500-e140-11e9-9373-f4b352b76b1a.png)

**Comment by Rexogamer** [comment]
When you say “users can’t read them”, are you suggesting the client would be laggy, or about screen size, or something else?

**Comment by chinesedfan** [comment]
> When you say “users can’t read them”, are you suggesting the client would be laggy, or about screen size, or something else?

Yes. I admit that the current behavior is not the best one. If we updated the API without UI enhancements to handle large files, users may face new problems.

**Comment by Rexogamer** [comment]
We don’t and won’t know what those issues are until we try the API change.
